test/unit/services/segment_delete_service_test.rb

```diff
@@ -44,6 +45,12 @@ def setup
       assert_includes segment_delete_service.errors[:base], 'reason of the error'
     end
 
+    test '#delete_user & #delete_user! do nothing and return nil when the feature is disabled' do
+      Features::SegmentDeletionConfig.stubs(enabled?: false)
+      assert_nil segment_delete_service.delete_user
```

Comment:
Does it have to return `nil` or could it be just `refute`?

Reply:
@guicassolato It is the way to know that it did nothing and not that it tried and went wrong 😄